### PR TITLE
bing + houndify stt

### PR DESCRIPTION
**Pull request by JarbasAI**
Adds bing + houndfy stt engines+ ibm test fix

### CODE REVIEW

**Comment by forslund** [comment]
Looks good, I'll see if I can try them out. Could you add test cases for these as well?

**Comment by JarbasAI** [comment]
honestly i didn't get api keys for them, just following the lib docs and noticed these were missing

idk if a version update may be needed in requirements.txt

what kind of test cases?

**Comment by forslund** [comment]
For these it seems like they work similar to existing STT backends so you should be able to copy existing testcase. For example:

https://github.com/MycroftAI/mycroft-core/blob/dev/test/unittests/stt/test_stt.py#L104

Basically ensuring that the correct speech_recognition methods gets called when invoking them and assuring no syntax errors occurs now or in the future.

File: mycroft/stt/__init__.py [hunk 1]
```diff
@@ -122,13 +122,35 @@ def get_response(self, response):
...
+
+
+class HoundifySTT(BasicSTT):
+    def __init__(self, lang="en-us", config=None):
+        super(HoundifySTT, self).__init__("houndify")
+
+    def execute(self, audio, language=None):
+        self.lang = language or self.lang
+        return self.recognizer.recognize_houndify(audio, self.username,
+                                                  self.password, self.lang)
```
Comment (forslund) [inline mycroft/stt/__init__.py]: Looking at the docs for houndify this doesn't seem quite right:

`recognize_houndify(self, audio_data, client_id, client_key, show_all=False)`

I think I would prefer `client_id` and `client_key` config parameters and not reuse username and password

File: mycroft/stt/__init__.py [hunk 2]
```diff
@@ -63,6 +63,15 @@ def __init__(self):
         self.password = str(self.credential.get("password"))
 
 
+class KeySTT(STT):
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply mycroft/stt/__init__.py]: is this a problem?

**Comment by forslund** [comment]
Finally tested this, and to what use. Works like a charm :) Merging

**Pull request merged**